src/hotspot/cpu/aarch64/vm_version_aarch64.cpp

```diff
@@ -46,6 +46,31 @@ int VM_Version::_dcache_line_size;
 int VM_Version::_icache_line_size;
 int VM_Version::_initial_sve_vector_length;
 
+PauseImplDesc VM_Version::_pause_impl_desc;
+
+static PauseImplDesc get_pause_impl_desc() {
+  const char *s = OnSpinWaitImpl;
+  unsigned int count = 1;
+  if (isdigit(*s)) {
+    count = *s - '0';
+    if (count == 0) {
+      vm_exit_during_initialization("Invalid value for OnSpinWaitImpl: zero instruction count", OnSpinWaitImpl);
+    }
+    s += 1;
+  }
```

Comment:
```suggestion
  while (isdigit(*s++));
  count = atoi(OnSpinWaitImpl);
```

Reply:
As far as I know, this combination of digits and named option is unusual in HotSpot; it may be unique. For the sake of not doing something so unfamiliar to our users, it may be worth separating the count and the option string.